lib/opam.ml

```diff
@@ -317,17 +317,21 @@ let opam_version () =
   let v =
     OS.Cmd.run_out Cmd.(cmd % "--version") |> OS.Cmd.out_string ~trim:true
   in
-  let of_str = function
-    | "1.2.2" -> `v1_2_2
-    | s -> (
-        match String.cut ~sep:"2." s with
-        | Some ("", _) -> `v2
-        | _ -> Fmt.failwith "opam: invalid version %s" s )
-  in
   match v with
-  | Ok (v, (_, `Exited 0)) -> of_str v
-  | Ok (_, (_, s)) -> Fmt.failwith "opam: %a" OS.Cmd.pp_status s
-  | Error (`Msg e) -> Fmt.failwith "opam: %s" e
+  | Ok (v, (_, `Exited 0)) when Bos_setup.String.is_prefix v ~affix:"1." ->
```

Comment:
How about extracting a parsing function for the output of `opam --version` here? That would also make the test cleaner as it doesn't cover much ground atm because it's testing the impure function.

Reply:
Agreed it's better, I improved it.